diff --git a/node_modules/@expo/config-plugins/build/ios/Swift.d.ts b/node_modules/@expo/config-plugins/build/ios/Swift.d.ts
index 1501a70..e66aac1 100644
--- a/node_modules/@expo/config-plugins/build/ios/Swift.d.ts
+++ b/node_modules/@expo/config-plugins/build/ios/Swift.d.ts
@@ -1,4 +1,3 @@
-/// <reference types="xcode" />
 import { ConfigPlugin, XcodeProject } from '../Plugin.types';
 /**
  * Ensure a Swift bridging header is created for the project.
diff --git a/node_modules/@expo/config-plugins/build/ios/XcodeProjectFile.d.ts b/node_modules/@expo/config-plugins/build/ios/XcodeProjectFile.d.ts
index 6c23a50..730336c 100644
--- a/node_modules/@expo/config-plugins/build/ios/XcodeProjectFile.d.ts
+++ b/node_modules/@expo/config-plugins/build/ios/XcodeProjectFile.d.ts
@@ -1,4 +1,3 @@
-/// <reference types="xcode" />
 import { ConfigPlugin, XcodeProject } from '../Plugin.types';
 /**
  * Create a build source file and link it to Xcode.
