diff --git a/node_modules/@gorhom/bottom-sheet/lib/typescript/components/bottomSheetModal/types.d.ts b/node_modules/@gorhom/bottom-sheet/lib/typescript/components/bottomSheetModal/types.d.ts
index 9e8e3ee..2a16d57 100644
--- a/node_modules/@gorhom/bottom-sheet/lib/typescript/components/bottomSheetModal/types.d.ts
+++ b/node_modules/@gorhom/bottom-sheet/lib/typescript/components/bottomSheetModal/types.d.ts
@@ -13,7 +13,7 @@ export type BottomSheetModalStackBehavior = keyof typeof MODAL_STACK_BEHAVIOR;
 
 // biome-ignore lint/suspicious/noExplicitAny: Using 'any' allows users to define their own strict types for 'data' property.
 export interface BottomSheetModalProps<T = any>
-  extends Omit<BottomSheetProps, 'containerHeight' | 'onClose'> {
+  extends Omit<BottomSheetProps, 'containerHeight' | 'onClose' | 'children'> {
   /**
    * Modal name to help identify the modal for later on.
    * @type string
diff --git a/node_modules/@gorhom/bottom-sheet/lib/typescript/components/bottomSheetScrollable/BottomSheetFlashList.d.ts b/node_modules/@gorhom/bottom-sheet/lib/typescript/components/bottomSheetScrollable/BottomSheetFlashList.d.ts
index e42b83e..4603deb 100644
--- a/node_modules/@gorhom/bottom-sheet/lib/typescript/components/bottomSheetScrollable/BottomSheetFlashList.d.ts
+++ b/node_modules/@gorhom/bottom-sheet/lib/typescript/components/bottomSheetScrollable/BottomSheetFlashList.d.ts
@@ -1,4 +1,4 @@
-import type { FlashListProps } from '@shopify/flash-list';
+type FlashListProps<T>  = any // import type { FlashListProps } from '@shopify/flash-list';
 import React, { type Ref } from 'react';
 import type Animated from 'react-native-reanimated';
 import type { BottomSheetScrollableProps } from './types';
diff --git a/node_modules/@gorhom/bottom-sheet/lib/typescript/types.d.ts b/node_modules/@gorhom/bottom-sheet/lib/typescript/types.d.ts
index 86c4e13..6203e28 100644
--- a/node_modules/@gorhom/bottom-sheet/lib/typescript/types.d.ts
+++ b/node_modules/@gorhom/bottom-sheet/lib/typescript/types.d.ts
@@ -190,7 +190,7 @@ export type ScrollEventsHandlersHookType = (
 };
 //#endregion
 
-export interface NullableAccessibilityProps extends AccessibilityProps {
+export interface NullableAccessibilityProps extends Omit<AccessibilityProps, 'accessible' | 'accessibilityLabel' | 'accessibilityHint' | 'accessibilityRole'> {
   accessible?: AccessibilityProps['accessible'] | null;
   accessibilityLabel?: AccessibilityProps['accessibilityLabel'] | null;
   accessibilityHint?: AccessibilityProps['accessibilityHint'] | null;
