diff --git a/node_modules/@khanacademy/simple-markdown/dist/es/index.js b/node_modules/@khanacademy/simple-markdown/dist/es/index.js
index 7d31dc5..1f76f2e 100644
--- a/node_modules/@khanacademy/simple-markdown/dist/es/index.js
+++ b/node_modules/@khanacademy/simple-markdown/dist/es/index.js
@@ -1,9 +1,7 @@
-import { addLibraryVersionToPerseusDebug } from '@khanacademy/perseus-core';
 
 // This file is processed by a Rollup plugin (replace) to inject the production
 const libName = "@khanacademy/simple-markdown";
 const libVersion = "0.11.2";
-addLibraryVersionToPerseusDebug(libName, libVersion);
 
 /* eslint-disable prefer-spread, no-regex-spaces, @typescript-eslint/no-unused-vars, guard-for-in, no-console, no-var */
 
diff --git a/node_modules/@khanacademy/simple-markdown/dist/index.d.ts b/node_modules/@khanacademy/simple-markdown/dist/index.d.ts
index 6f3c7fc..3221b28 100644
--- a/node_modules/@khanacademy/simple-markdown/dist/index.d.ts
+++ b/node_modules/@khanacademy/simple-markdown/dist/index.d.ts
@@ -60,32 +60,32 @@ type HtmlOutputRule = {
     readonly html: HtmlNodeOutput | null;
 };
 type ArrayRule = {
-    readonly react?: ArrayNodeOutput<ReactElements>;
-    readonly html?: ArrayNodeOutput<string>;
+    //readonly react?: ArrayNodeOutput<ReactElements>;
+    //readonly html?: ArrayNodeOutput<string>;
     readonly [key: string]: ArrayNodeOutput<any>;
 };
 type ParserRules = {
-    readonly Array?: ArrayRule;
+    //readonly Array?: ArrayRule;
     readonly [type: string]: ParserRule;
 };
 type OutputRules<Rule> = {
-    readonly Array?: ArrayRule;
+    //readonly Array?: ArrayRule;
     readonly [type: string]: Rule;
 };
 type Rules<OutputRule> = {
-    readonly Array?: ArrayRule;
+    //readonly Array?: ArrayRule;
     readonly [type: string]: ParserRule & OutputRule;
 };
 type ReactRules = {
-    readonly Array?: {
-        readonly react: ArrayNodeOutput<ReactElements>;
-    };
+    //readonly Array?: {
+     //   readonly react: ArrayNodeOutput<ReactElements>;
+    //};
     readonly [type: string]: ParserRule & ReactOutputRule;
 };
 type HtmlRules = {
-    readonly Array?: {
-        readonly html: ArrayNodeOutput<string>;
-    };
+    //readonly Array?: {
+     //   readonly html: ArrayNodeOutput<string>;
+    //};
     readonly [type: string]: ParserRule & HtmlOutputRule;
 };
 type NonNullReactOutputRule = {
diff --git a/node_modules/@khanacademy/simple-markdown/dist/index.js b/node_modules/@khanacademy/simple-markdown/dist/index.js
index d0e2c4a..9573e7c 100644
--- a/node_modules/@khanacademy/simple-markdown/dist/index.js
+++ b/node_modules/@khanacademy/simple-markdown/dist/index.js
@@ -2,12 +2,10 @@
 
 Object.defineProperty(exports, '__esModule', { value: true });
 
-var perseusCore = require('@khanacademy/perseus-core');
 
 // This file is processed by a Rollup plugin (replace) to inject the production
 const libName = "@khanacademy/simple-markdown";
 const libVersion = "0.11.2";
-perseusCore.addLibraryVersionToPerseusDebug(libName, libVersion);
 
 /* eslint-disable prefer-spread, no-regex-spaces, @typescript-eslint/no-unused-vars, guard-for-in, no-console, no-var */
 
