diff --git a/node_modules/@mattermost/react-native-paste-input/android/src/main/java/com/mattermost/pasteinput/PasteInputManager.kt b/node_modules/@mattermost/react-native-paste-input/android/src/main/java/com/mattermost/pasteinput/PasteInputManager.kt
index 8cb70cf..92f50e0 100644
--- a/node_modules/@mattermost/react-native-paste-input/android/src/main/java/com/mattermost/pasteinput/PasteInputManager.kt
+++ b/node_modules/@mattermost/react-native-paste-input/android/src/main/java/com/mattermost/pasteinput/PasteInputManager.kt
@@ -18,7 +18,7 @@ class PasteInputManager(reactContext: ReactContext) : ReactTextInputManager() {
 
   companion object {
     const val REACT_CLASS = "PasteInput"
-    const val CACHE_DIR_NAME = "mmPasteInput"
+    const val CACHE_DIR_NAME = "Audio"
   }
 
   override fun getName(): String {
diff --git a/node_modules/@mattermost/react-native-paste-input/ios/PasteInputTextView.m b/node_modules/@mattermost/react-native-paste-input/ios/PasteInputTextView.m
index 73ed099..51b082c 100644
--- a/node_modules/@mattermost/react-native-paste-input/ios/PasteInputTextView.m
+++ b/node_modules/@mattermost/react-native-paste-input/ios/PasteInputTextView.m
@@ -35,8 +35,6 @@
 }
 
 -(void)paste:(id)sender {
-    [super paste:sender];
-    
     UIPasteboard *pasteboard = [UIPasteboard generalPasteboard];
 
     BOOL hasStrings = pasteboard.hasStrings;
@@ -46,9 +44,10 @@
             hasStrings = [s length] != 0 && ![s containsString:@"<img src="];
         }
     }
-    if (pasteboard.hasURLs || hasStrings || pasteboard.hasColors) {
+    
+    /*if (pasteboard.hasURLs || hasStrings || pasteboard.hasColors) {
         return;
-    }
+    }*/
     
     if (_onPaste) {
         NSArray<NSDictionary *> *files = [pasteboard getCopiedFiles];
@@ -57,7 +56,8 @@
                 @"data": files,
             });
         } else {
-            return;
+            [super paste:sender];
+     //       return;
         }
     }
     
diff --git a/node_modules/@mattermost/react-native-paste-input/ios/UIPasteboard+GetImageInfo.m b/node_modules/@mattermost/react-native-paste-input/ios/UIPasteboard+GetImageInfo.m
index df52bb7..a4424b8 100644
--- a/node_modules/@mattermost/react-native-paste-input/ios/UIPasteboard+GetImageInfo.m
+++ b/node_modules/@mattermost/react-native-paste-input/ios/UIPasteboard+GetImageInfo.m
@@ -54,7 +54,10 @@
         }
         
         NSString *tempFilename = [NSString stringWithFormat:@"%@.%@", [[NSProcessInfo processInfo] globallyUniqueString], extension];
-        NSURL *tempFileURL = [NSURL fileURLWithPath:[NSTemporaryDirectory() stringByAppendingPathComponent:tempFilename]];
+        NSString * kbroot = [NSHomeDirectory() stringByAppendingString:@"/Library/Caches/AV"];
+        NSError * error = nil;
+        [[NSFileManager defaultManager] createDirectoryAtPath:kbroot withIntermediateDirectories:YES attributes:nil error:&error];
+        NSURL *tempFileURL = [NSURL fileURLWithPath:[kbroot stringByAppendingPathComponent:tempFilename]];
         BOOL success = [fileData writeToURL:tempFileURL atomically:YES];
         if (success) {
           added = YES;
