diff --git a/node_modules/@react-spring/types/util.d.ts b/node_modules/@react-spring/types/util.d.ts
index 57c0773..9bd1a73 100644
--- a/node_modules/@react-spring/types/util.d.ts
+++ b/node_modules/@react-spring/types/util.d.ts
@@ -118,13 +118,13 @@ type Intersect<U> = (U extends any ? (k: U) => void : never) extends (
   ? I
   : never
 
-export type Exclusive<T> = AllKeys<T> extends infer K
+export type Exclusive<T> = any/*AllKeys<T> extends infer K
   ? T extends any
     ? Remap<
         LoosePick<T, K> & { [P in Exclude<K & keyof any, keyof T>]?: undefined }
       >
     : never
-  : never
+  : never*/
 /** An object that needs to be manually disposed of */
 export interface Disposable {
   dispose(): void
