diff --git a/node_modules/@react-navigation/bottom-tabs/lib/module/unstable/NativeBottomTabView.native.js b/node_modules/@react-navigation/bottom-tabs/lib/module/unstable/NativeBottomTabView.native.js
index dfcee2b..dd1b37a 100644
--- a/node_modules/@react-navigation/bottom-tabs/lib/module/unstable/NativeBottomTabView.native.js
+++ b/node_modules/@react-navigation/bottom-tabs/lib/module/unstable/NativeBottomTabView.native.js
@@ -23,6 +23,10 @@ export function NativeBottomTabView({
   } = useTheme();
   const focusedRouteKey = state.routes[state.index].key;
   const previousRouteKeyRef = React.useRef(focusedRouteKey);
+  const [navStateRequest, setNavStateRequest] = React.useState(() => ({
+    selectedScreenKey: state.routes[state.index].key,
+    baseProvenance: 0,
+  }));
   React.useEffect(() => {
     const previousRouteKey = previousRouteKeyRef.current;
     if (previousRouteKey !== focusedRouteKey && descriptors[previousRouteKey]?.options.popToTopOnBlur) {
@@ -109,9 +113,14 @@ export function NativeBottomTabView({
       tabBarItemActiveIndicatorEnabled: currentOptions?.tabBarActiveIndicatorEnabled,
       tabBarItemRippleColor: currentOptions?.tabBarRippleColor,
       experimentalControlNavigationStateInJS: false,
-      onNativeFocusChange: e => {
-        const route = state.routes.find(route => route.key === e.nativeEvent.tabKey);
+      navStateRequest: navStateRequest,
+      onTabSelected: e => {
+        const route = state.routes.find(route => route.key === e.nativeEvent.selectedScreenKey);
         if (route) {
+          setNavStateRequest({
+            selectedScreenKey: e.nativeEvent.selectedScreenKey,
+            baseProvenance: e.nativeEvent.provenance,
+          });
           navigation.emit({
             type: 'tabPress',
             target: route.key
@@ -190,9 +199,8 @@ export function NativeBottomTabView({
             closing: true,
             route
           }),
-          tabKey: route.key,
-          icon: icon,
-          selectedIcon: selectedIcon?.ios ?? selectedIcon?.shared,
+          screenKey: route.key,
+          ios: {icon: icon?.ios, selectedIcon: selectedIcon?.ios ?? selectedIcon?.shared},
           tabBarItemBadgeBackgroundColor: badgeBackgroundColor,
           tabBarItemBadgeTextColor: badgeTextColor,
           badgeValue: tabBarBadge?.toString(),
