diff --git a/node_modules/@react-navigation/elements/lib/module/useFrameSize.js b/node_modules/@react-navigation/elements/lib/module/useFrameSize.js
index af9105b..6b01a9b 100644
--- a/node_modules/@react-navigation/elements/lib/module/useFrameSize.js
+++ b/node_modules/@react-navigation/elements/lib/module/useFrameSize.js
@@ -11,7 +11,7 @@ import { useSyncExternalStoreWithSelector } from 'use-sync-external-store/with-s
 // Load with require to avoid error from webpack due to missing export in older versions
 // eslint-disable-next-line import-x/no-commonjs
 import { jsx as _jsx, Fragment as _Fragment, jsxs as _jsxs } from "react/jsx-runtime";
-const SafeAreaListener = require('react-native-safe-area-context').SafeAreaListener;
+import { SafeAreaListener } from 'react-native-safe-area-context';
 const FrameContext = /*#__PURE__*/React.createContext(undefined);
 export function useFrameSize(selector, throttle) {
   const context = React.useContext(FrameContext);
