diff --git a/node_modules/@react-native-picker/picker/typings/Picker.d.ts b/node_modules/@react-native-picker/picker/typings/Picker.d.ts
index 0e21756..a51c567 100644
--- a/node_modules/@react-native-picker/picker/typings/Picker.d.ts
+++ b/node_modules/@react-native-picker/picker/typings/Picker.d.ts
@@ -132,7 +132,7 @@ declare class Picker<T> extends React.Component<PickerProps<T>, {}> {
    */
   static readonly MODE_DROPDOWN: 'dropdown';
 
-  static Item: React.ComponentType<PickerItemProps<T>>;
+  static Item: React.ComponentType<PickerItemProps<any>>;
 
   /**
    * @platform android
