diff --git a/node_modules/@shopify/flash-list/dist/FlashList.js b/node_modules/@shopify/flash-list/dist/FlashList.js
index ea05c45..becc75d 100644
--- a/node_modules/@shopify/flash-list/dist/FlashList.js
+++ b/node_modules/@shopify/flash-list/dist/FlashList.js
@@ -24,7 +24,7 @@ var FlashList = /** @class */ (function (_super) {
         var _a;
         _this = _super.call(this, props) || this;
         _this.listFixedDimensionSize = 0;
-        _this.transformStyle = { transform: [{ scaleY: -1 }] };
+        _this.transformStyle = react_native_1.Platform.OS === 'ios' ? { transform: [{ scaleY: -1 }] } : {scaleY: -1};
         _this.transformStyleHorizontal = { transform: [{ scaleX: -1 }] };
         _this.distanceFromWindow = 0;
         _this.contentStyle = {
diff --git a/node_modules/@shopify/flash-list/src/FlashList.tsx b/node_modules/@shopify/flash-list/src/FlashList.tsx
index 5ec2c34..35ae9e8 100644
--- a/node_modules/@shopify/flash-list/src/FlashList.tsx
+++ b/node_modules/@shopify/flash-list/src/FlashList.tsx
@@ -5,6 +5,7 @@ import {
   LayoutChangeEvent,
   NativeSyntheticEvent,
   NativeScrollEvent,
+  Platform,
 } from "react-native";
 import {
   BaseItemAnimator,
@@ -68,7 +69,7 @@ class FlashList<T> extends React.PureComponent<
   private rlvRef?: RecyclerListView<RecyclerListViewProps, any>;
   private stickyContentContainerRef?: PureComponentWrapper;
   private listFixedDimensionSize = 0; 
-  private transformStyle = { transform: [{ scaleY: -1 }] };
+  private transformStyle = Platform.OS === 'ios' ? { transform: [{ scaleY: -1 }] } : {scaleY: -1};
   private transformStyleHorizontal = { transform: [{ scaleX: -1 }] };
   private distanceFromWindow = 0;
   private contentStyle: ContentStyleExplicit = {
