diff --git a/node_modules/@shopify/flash-list/dist/FlashList.js b/node_modules/@shopify/flash-list/dist/FlashList.js
index 86495d4..7079c46 100644
--- a/node_modules/@shopify/flash-list/dist/FlashList.js
+++ b/node_modules/@shopify/flash-list/dist/FlashList.js
@@ -109,14 +109,15 @@ var FlashList = /** @class */ (function (_super) {
             }
         };
         _this.separator = function (index) {
+	    const offset = this.props.inverted ? -1 : 1
             // Make sure we have data and don't read out of bounds
             if (_this.props.data === null ||
                 _this.props.data === undefined ||
-                index + 1 >= _this.props.data.length) {
+		index + offset >= _this.props.data.length) {
                 return null;
             }
             var leadingItem = _this.props.data[index];
-            var trailingItem = _this.props.data[index + 1];
+	    var trailingItem = _this.props.data[index + offset];
             var props = {
                 leadingItem: leadingItem,
                 trailingItem: trailingItem,
