diff --git a/node_modules/electron/electron.d.ts b/node_modules/electron/electron.d.ts
index 630b407..765e74d 100644
--- a/node_modules/electron/electron.d.ts
+++ b/node_modules/electron/electron.d.ts
@@ -1768,7 +1768,7 @@ declare namespace Electron {
      *
      * @platform darwin
      */
-    readonly dock: Dock;
+    readonly dock: Dock | undefined;
     /**
      * A `boolean` property that returns  `true` if the app is packaged, `false`
      * otherwise. For many apps, this property can be used to distinguish development
