diff --git a/node_modules/electron/electron.d.ts b/node_modules/electron/electron.d.ts
index ed11e91..f22b138 100644
--- a/node_modules/electron/electron.d.ts
+++ b/node_modules/electron/electron.d.ts
@@ -1833,7 +1833,7 @@ declare namespace Electron {
      *
      * @platform darwin
      */
-    readonly dock: Dock;
+    readonly dock: Dock|undefined;
     /**
      * A `boolean` property that returns  `true` if the app is packaged, `false`
      * otherwise. For many apps, this property can be used to distinguish development
