diff --git a/node_modules/expo-image-picker/ios/ImagePickerModule.swift b/node_modules/expo-image-picker/ios/ImagePickerModule.swift
index 08264ed..71a85f9 100644
--- a/node_modules/expo-image-picker/ios/ImagePickerModule.swift
+++ b/node_modules/expo-image-picker/ios/ImagePickerModule.swift
@@ -103,7 +103,7 @@ public class ImagePickerModule: Module, OnMediaPickingResultHandler {
     }
 
     picker.mediaTypes = options.mediaTypes.toArray()
-    picker.videoExportPreset = options.videoExportPreset.toAVAssetExportPreset()
+    //picker.videoExportPreset = options.videoExportPreset.toAVAssetExportPreset()
     picker.videoQuality = options.videoQuality.toQualityType()
     picker.videoMaximumDuration = options.videoMaxDuration
     picker.modalPresentationStyle = options.presentationStyle.toPresentationStyle()
