diff --git a/node_modules/expo-image/ios/ImageView.swift b/node_modules/expo-image/ios/ImageView.swift
index bb6f6a5..ca6a40f 100644
--- a/node_modules/expo-image/ios/ImageView.swift
+++ b/node_modules/expo-image/ios/ImageView.swift
@@ -117,14 +117,9 @@ public final class ImageView: ExpoView {
     addSubview(sdImageView)
   }
 
-  public override func didMoveToWindow() {
-    if window == nil {
-      // Cancel pending requests when the view is unmounted.
-      cancelPendingOperation()
-      return
-    }
-
-    loadPlaceholderIfNecessary()
+  deinit {
+    // Cancel pending requests when the view is deallocated.
+    cancelPendingOperation()
   }
 
   public override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
