diff --git a/node_modules/expo-image-picker/ios/ImagePickerModule.swift b/node_modules/expo-image-picker/ios/ImagePickerModule.swift
index e7de67a..a859b98 100644
--- a/node_modules/expo-image-picker/ios/ImagePickerModule.swift
+++ b/node_modules/expo-image-picker/ios/ImagePickerModule.swift
@@ -128,7 +128,7 @@ public class ImagePickerModule: Module, OnMediaPickingResultHandler {
       }
     }
 
-    picker.videoExportPreset = options.videoExportPreset.toAVAssetExportPreset()
+    // picker.videoExportPreset = options.videoExportPreset.toAVAssetExportPreset()
     picker.videoQuality = options.videoQuality.toQualityType()
     picker.videoMaximumDuration = options.videoMaxDuration
 
