diff --git a/node_modules/lottie-react-native/lottie-react-native.podspec b/node_modules/lottie-react-native/lottie-react-native.podspec
index d2c5424..de6aeb6 100644
--- a/node_modules/lottie-react-native/lottie-react-native.podspec
+++ b/node_modules/lottie-react-native/lottie-react-native.podspec
@@ -22,5 +22,5 @@ Pod::Spec.new do |s|
   s.pod_target_xcconfig = { 'DEFINES_MODULE' => 'YES' }
   s.swift_version = "5.0"
   s.dependency 'React-Core'
-  s.dependency 'lottie-ios', '~> 3.4.0'
+  s.dependency 'lottie-ios', '~> 3.5.0'
 end
