diff --git a/node_modules/menubar/lib/Menubar.js b/node_modules/menubar/lib/Menubar.js
index 9155dcc..44ffa72 100644
--- a/node_modules/menubar/lib/Menubar.js
+++ b/node_modules/menubar/lib/Menubar.js
@@ -359,7 +359,8 @@ var Menubar = /** @class */ (function (_super) {
                         if (this._options.showOnAllWorkspaces !== false) {
                             // https://github.com/electron/electron/issues/37832#issuecomment-1497882944
                             this._browserWindow.setVisibleOnAllWorkspaces(true, {
-                                skipTransformProcessType: true, // Avoid damaging the original visible state of app.dock
+			        visibleOnFullScreen: true,
+                                //skipTransformProcessType: true, // Avoid damaging the original visible state of app.dock
                             });
                         }
                         this._browserWindow.on('close', this.windowClear.bind(this));
