diff --git a/node_modules/react-native-zoom-toolkit/src/commons/hooks/useDoubleTapCommons.ts b/node_modules/react-native-zoom-toolkit/src/commons/hooks/useDoubleTapCommons.ts
index 236c4fc..67e43db 100644
--- a/node_modules/react-native-zoom-toolkit/src/commons/hooks/useDoubleTapCommons.ts
+++ b/node_modules/react-native-zoom-toolkit/src/commons/hooks/useDoubleTapCommons.ts
@@ -36,8 +36,8 @@ export const useDoubleTapCommons = ({
 
     const originX = event.x - container.width.value / 2;
     const originY = event.y - container.height.value / 2;
-    const toScale =
-      scale.value >= maxScale.value * 0.8 ? minScale : maxScale.value;
+    const toScale = scale.value === minScale ? maxScale.value : minScale;
+      //scale.value >= maxScale.value * 0.8 ? minScale : maxScale.value;
 
     const { x, y } = pinchTransform({
       toScale: toScale,
