diff --git a/node_modules/react-native-screens/common/cpp/react/renderer/components/rnscreens/RNSModalScreenShadowNode.h b/node_modules/react-native-screens/common/cpp/react/renderer/components/rnscreens/RNSModalScreenShadowNode.h
index af1a193..9bf7cec 100644
--- a/node_modules/react-native-screens/common/cpp/react/renderer/components/rnscreens/RNSModalScreenShadowNode.h
+++ b/node_modules/react-native-screens/common/cpp/react/renderer/components/rnscreens/RNSModalScreenShadowNode.h
@@ -20,7 +20,7 @@ class JSI_EXPORT RNSModalScreenShadowNode final
  public:
   using ConcreteViewShadowNode::ConcreteViewShadowNode;
 
-  Point getContentOriginOffset(bool includeTransform) const override;
+  Point getContentOriginOffset(bool includeTransform) const;
   static ShadowNodeTraits BaseTraits() {
     auto traits = ConcreteViewShadowNode::BaseTraits();
     traits.set(ShadowNodeTraits::Trait::RootNodeKind);
diff --git a/node_modules/react-native-screens/common/cpp/react/renderer/components/rnscreens/RNSScreenShadowNode.h b/node_modules/react-native-screens/common/cpp/react/renderer/components/rnscreens/RNSScreenShadowNode.h
index a6c5dc0..45f0a1f 100644
--- a/node_modules/react-native-screens/common/cpp/react/renderer/components/rnscreens/RNSScreenShadowNode.h
+++ b/node_modules/react-native-screens/common/cpp/react/renderer/components/rnscreens/RNSScreenShadowNode.h
@@ -26,7 +26,7 @@ class JSI_EXPORT RNSScreenShadowNode final : public ConcreteViewShadowNode<
 
 #pragma mark - ShadowNode overrides
 
-  Point getContentOriginOffset(bool includeTransform) const override;
+  Point getContentOriginOffset(bool includeTransform) const;
 
   void layout(LayoutContext layoutContext) override;
 
diff --git a/node_modules/react-native-screens/lib/typescript/native-stack/types.d.ts b/node_modules/react-native-screens/lib/typescript/native-stack/types.d.ts
index ce3a88f..ae7da85 100644
--- a/node_modules/react-native-screens/lib/typescript/native-stack/types.d.ts
+++ b/node_modules/react-native-screens/lib/typescript/native-stack/types.d.ts
@@ -49,7 +49,7 @@ export type NativeStackNavigationEventMap = {
         };
     };
 };
-export type NativeStackNavigationProp<ParamList extends ParamListBase, RouteName extends keyof ParamList = string> = NavigationProp<ParamList, RouteName, StackNavigationState<ParamList>, NativeStackNavigationOptions, NativeStackNavigationEventMap> & StackActionHelpers<ParamList>;
+export type NativeStackNavigationProp<ParamList extends ParamListBase, RouteName extends keyof ParamList = string> = any // NavigationProp<ParamList, RouteName, StackNavigationState<ParamList>, NativeStackNavigationOptions, NativeStackNavigationEventMap> & StackActionHelpers<ParamList>;
 export type NativeStackScreenProps<ParamList extends ParamListBase, RouteName extends keyof ParamList = string> = {
     navigation: NativeStackNavigationProp<ParamList, RouteName>;
     route: RouteProp<ParamList, RouteName>;
@@ -461,8 +461,8 @@ export type NativeStackNavigationOptions = {
     transitionAnimation?: AnimatedScreenTransition;
     screenEdgeGesture?: boolean;
 };
-export type NativeStackNavigatorProps = DefaultNavigatorOptions<NativeStackNavigationOptions> & StackRouterOptions & NativeStackNavigationConfig;
-export type NativeStackDescriptor = Descriptor<ParamListBase, string, StackNavigationState<ParamListBase>, NativeStackNavigationOptions>;
+export type NativeStackNavigatorProps = any // DefaultNavigatorOptions<NativeStackNavigationOptions> & StackRouterOptions & NativeStackNavigationConfig;
+export type NativeStackDescriptor = any // Descriptor<ParamListBase, string, StackNavigationState<ParamListBase>, NativeStackNavigationOptions>;
 export type NativeStackDescriptorMap = {
     [key: string]: NativeStackDescriptor;
 };
