diff --git a/node_modules/react-native-safe-area-context/lib/typescript/src/specs/NativeSafeAreaView.d.ts b/node_modules/react-native-safe-area-context/lib/typescript/src/specs/NativeSafeAreaView.d.ts
index fcd88fc..1a5cf0c 100644
--- a/node_modules/react-native-safe-area-context/lib/typescript/src/specs/NativeSafeAreaView.d.ts
+++ b/node_modules/react-native-safe-area-context/lib/typescript/src/specs/NativeSafeAreaView.d.ts
@@ -1,4 +1,3 @@
-/// <reference types="react-native/types/modules/codegen" />
 import type { WithDefault } from 'react-native/Libraries/Types/CodegenTypes';
 import type { ViewProps, HostComponent } from 'react-native';
 export interface NativeProps extends ViewProps {
