diff --git a/node_modules/react-native-gesture-handler/apple/RNGestureHandlerRegistry.m b/node_modules/react-native-gesture-handler/apple/RNGestureHandlerRegistry.m
index 0c52243..9e7e4b9 100644
--- a/node_modules/react-native-gesture-handler/apple/RNGestureHandlerRegistry.m
+++ b/node_modules/react-native-gesture-handler/apple/RNGestureHandlerRegistry.m
@@ -52,7 +52,9 @@ - (void)attachHandlerWithTag:(NSNumber *)handlerTag
     handler = _handlers[handlerTag];
   }
 
-  RCTAssert(handler != nil, @"Handler for tag %@ does not exists", handlerTag);
+  if (handler == nil) {
+    return;
+  }
   [handler unbindFromView];
   handler.actionType = actionType;
   [handler bindToView:view];
