diff --git a/node_modules/react-native-reanimated/src/reanimated2/PlatformChecker.ts b/node_modules/react-native-reanimated/src/reanimated2/PlatformChecker.ts
index 9b3fcb1..327cf21 100644
--- a/node_modules/react-native-reanimated/src/reanimated2/PlatformChecker.ts
+++ b/node_modules/react-native-reanimated/src/reanimated2/PlatformChecker.ts
@@ -44,6 +44,7 @@ export function isWindowAvailable() {
 }
 
 export function isReducedMotion() {
+return false;
   return isWeb()
     ? isWindowAvailable()
       ? // @ts-ignore Fallback if `window` is undefined.
