diff --git a/node_modules/react-native/Libraries/Animated/Animated.d.ts b/node_modules/react-native/Libraries/Animated/Animated.d.ts
index aeeb981..5d96b30 100644
--- a/node_modules/react-native/Libraries/Animated/Animated.d.ts
+++ b/node_modules/react-native/Libraries/Animated/Animated.d.ts
@@ -100,7 +100,7 @@ export namespace Animated {
   }
 
   class AnimatedInterpolation<
-    OutputT extends number | string,
+    OutputT extends number | string = number,
   > extends AnimatedWithChildren {
     interpolate(
       config: InterpolationConfigType,
diff --git a/node_modules/react-native/types/index.d.ts b/node_modules/react-native/types/index.d.ts
index 5177707..e2c893e 100644
--- a/node_modules/react-native/types/index.d.ts
+++ b/node_modules/react-native/types/index.d.ts
@@ -68,7 +68,6 @@
 /// <reference path="modules/BatchedBridge.d.ts" />
 /// <reference path="modules/Codegen.d.ts" />
 /// <reference path="modules/Devtools.d.ts" />
-/// <reference path="modules/globals.d.ts" />
 /// <reference path="modules/LaunchScreen.d.ts" />
 
 export * from '../Libraries/ActionSheetIOS/ActionSheetIOS';
