diff --git a/node_modules/zustand/esm/react.mjs b/node_modules/zustand/esm/react.mjs
index c247a0d..3af8217 100644
--- a/node_modules/zustand/esm/react.mjs
+++ b/node_modules/zustand/esm/react.mjs
@@ -1,4 +1,4 @@
-import ReactExports from 'react';
+import * as ReactExports from 'react';
 import { createStore } from 'zustand/vanilla';
 
 const { useDebugValue, useSyncExternalStore } = ReactExports;
diff --git a/node_modules/zustand/esm/react/shallow.mjs b/node_modules/zustand/esm/react/shallow.mjs
index e07a27e..98dba8c 100644
--- a/node_modules/zustand/esm/react/shallow.mjs
+++ b/node_modules/zustand/esm/react/shallow.mjs
@@ -1,4 +1,4 @@
-import ReactExports from 'react';
+import * as ReactExports from 'react';
 import { shallow } from 'zustand/vanilla/shallow';
 
 const { useRef } = ReactExports;
